toolchain/lower/file_context.h

```diff
@@ -219,6 +294,34 @@ class FileContext {
 
   // Global format string for `printf.int.format` used by the PrintInt builtin.
   llvm::Value* printf_int_format_string_ = nullptr;
+
+  // For a generic function, keep track of the specifics for which LLVM
+  // function declarations were created. Those can be retrieved then via
+  // from specific_functions_, via specific_functions_[specific_id.index].
+  // We resize this to the correct size. Indexed by generic_id.index.
+  llvm::SmallVector<llvm::SmallVector<SemIR::SpecificId>, 0> lowered_specifics_;
+
+  // For specifics that exist in lowered_specifics, a hash of their function
+  // type information: return and parameter types. We resize this to the
+  // correct size. Indexed by specific_id.index.
```

Comment:
There are a lot of things indexed by `SpecificId`: had you considered having a single vector of all the items that need to be tied to a given specific, rather than a separate vector for each?

Reply:
When each of these are used differs enough to keep them separate for now I think - but adding this note to a subsequent refactoring change (i.e. factoring out the coalescing logic).